include/dd/Package.hpp

```diff
@@ -303,6 +303,32 @@ namespace dd {
             return f;
         }
 
+        // generate the decision diagram from an arbitrary state vector
+        vEdge makeStateFromVector(const CVec& stateVector) {
+            if (stateVector.empty()) {
+                return vEdge::one;
+            }
+            const auto& length = stateVector.size();
+            if ((length & (length - 1)) != 0) {
+                throw std::invalid_argument("State vector must have a length of a power of two.");
+            }
+
+            [[maybe_unused]] const auto before = cn.cacheCount();
+
+            const auto level = static_cast<std::size_t>(std::log2(length)) - 1;
```

Comment:
Could be casted to `Qubit` type.

Reply:
Also done in https://github.com/cda-tum/dd_package/pull/135/commits/d59548b5625470658fe1d982b6aaa4bfd981148e